src/pool/srv_pool_chkpt.c

```diff
@@ -44,9 +60,117 @@ wait_fn(struct chkpt_ctx *ctx)
 static void
 wake_fn(struct chkpt_ctx *ctx)
 {
+	ctx->cc_waiting = 0;
```

Comment:
if nowhere else is using these two functions (wait_fn, wake_fn), then we probably just call eventual APIs directly and remove these functions.